Scripts/Proxies/ProxyNode.cs

```diff
@@ -106,76 +124,58 @@ public VisibilityState(Renderer renderer, AffordanceTooltip[] tooltips, Affordan
                     {
                         foreach (var kvp in m_Visibilities)
                         {
-                            definition = kvp.Value.definition;
+                            visibilityState = kvp.Value;
+                            definition = visibilityState.definition;
                             break;
                         }
                     }
 
-                    if (visible != m_WasVisible)
-                        m_VisibleChangeTime = time;
-
-                    var timeDiff = time - m_VisibleChangeTime;
                     var fadeDuration = visible ? fadeInDuration : fadeOutDuration;
-
                     switch (definition.visibilityType)
                     {
                         case VisibilityControlType.AlphaProperty:
-                            var alphaProperty = definition.alphaProperty;
-                            if (visible != m_WasVisible)
-                                m_StartColor = material.GetFloat(alphaProperty) * Color.white;
+                            Action<float> setFloat = null;
+                            if (visibilityState != null)
```

Comment:
Would we be better served just making a visibility state here rather than a setfloat function that is essentially the same thing as visibilitystate's setfloat?

Reply:
I went back and forth on this. Ultimately decided the overhead of the unused variables wasn't worth it, but it would be cleaner to read. I'll give this a shot.